lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenHoverParticipant.java

```diff
@@ -146,40 +145,98 @@ private String collectArtifactDescription(IHoverRequest request, boolean isPlugi
+		PluginDescriptor pluginDescriptor = MavenPluginUtils.getContainingPluginDescriptor(request, cache,
+				pluginManager);
+		if (pluginDescriptor != null) {
 			for (MojoDescriptor mojo : pluginDescriptor.getMojos()) {
 				if (!node.getChild(0).getNodeValue().trim().isEmpty() && node.hasChildNodes()
 						&& node.getChild(0).getNodeValue().equals(mojo.getGoal())) {
 					return mojo.getDescription();
 				}
-			}			
+			}
 		}
 		return null;
 	}
 
 	private String collectPuginConfiguration(IPositionRequest request) {
-		List<Parameter> parameters = MavenPluginUtils.collectPluginConfigurationParameters(request, cache, pluginManager);
+		List<Parameter> parameters = MavenPluginUtils.collectPluginConfigurationParameters(request, cache,
+				pluginManager);
 		DOMNode node = request.getNode();
-		
+
 		for (Parameter parameter : parameters) {
 			if (node.getLocalName().equals(parameter.getName())) {
 				return MavenPluginUtils.getMarkupDescription(parameter).getValue();
 			}
 		}
 		return null;
 	}
-	
-	private Hover toHover(String description) {
-		Hover hover = new Hover();
-		hover.setContents(new MarkupContent("plaintext", description));
-		return hover;
-	}
 
 	@Override
 	public String onText(IHoverRequest request) throws Exception {
-		// TODO Auto-generated method stub
+		String mavenProperty = getMavenPropertyInHover(request);
+		if (mavenProperty != null) {
+			return collectProperty(request, mavenProperty);
+		}
+		
 		return null;
 	}
 	
+	public String getMavenPropertyInHover(IPositionRequest request) {
+		DOMNode tag = request.getNode();
+		String tagText = tag.getNodeValue();
+		
+		int hoverLocation = request.getOffset();
+		int propertyOffset = request.getNode().getStart();
+		int beforeHover = hoverLocation - propertyOffset;
+		
+		String beforeHoverText = tagText.substring(0, beforeHover);
+		String afterHoverText = tagText.substring(beforeHover);
+		
+		int indexOpen = beforeHoverText.lastIndexOf("${");
+		int indexCloseBefore = beforeHoverText.lastIndexOf("}");
+		int indexCloseAfter = afterHoverText.indexOf("}");
+		if (indexOpen > indexCloseBefore) {
+			return tagText.substring(indexOpen + 2, indexCloseAfter + beforeHover);
+		}
+		
+		
+		return null;
+	}
+	
+	private String collectProperty(IPositionRequest request, String property) {
+		DOMDocument doc = request.getXMLDocument();
+		MavenProject project = cache.getLastSuccessfulMavenProject(doc);
+		if (project != null) {
+			// TODO: Factor this with code from MavenCompletionParticipant.completeProperties()
+			Map<String, String> allProps = getMavenProjectProperties(project);
+			
+			for (Entry<String, String> prop : allProps.entrySet()) {
+				String mavenProperty = prop.getKey();
+				if (property.equals(mavenProperty)) {
+					return "Property: " + mavenProperty + "\n" + "Value: " +  prop.getValue() + "\n";
+				}
+			}
+		}
+		return null;
+	}
+
+	public static Map<String, String> getMavenProjectProperties(MavenProject project) {
```

Comment:
This belongs in a Utility class since it's need is not unique to MavenHoverParticipant.
Couldn't find a suitable class to move it to, perhaps a new one should be made?